src/prefect/agent/kubernetes/agent.py

```diff
@@ -103,9 +104,99 @@ def __init__(
             config.load_kube_config()
 
         self.batch_client = client.BatchV1Api()
+        self.core_client = client.CoreV1Api()
+        self.k8s_client = client
 
         self.logger.debug(f"Namespace: {self.namespace}")
 
+    def manage_jobs(self) -> None:
+        """
+        This function checks if jobs are `Failed` or `Succeeded` and if they are then the jobs are
+        deleted from the namespace. If one of the job's pods happen to run into image pulling errors
+        then the flow run is failed and the job is still deleted.
+        """
+        self.logger.debug(
+            "Retrieving information of jobs that are currently in the cluster..."
+        )
+
+        more = True
+        _continue = ""
+        while more:
+            try:
+                jobs = self.batch_client.list_namespaced_job(
+                    namespace=self.namespace,
+                    label_selector="prefect.io/identifier",
+                    limit=20,
+                    _continue=_continue,
+                )
+                _continue = jobs.metadata._continue
+                more = bool(_continue)
+
+                for job in jobs.items:
+                    delete_job = False
```

Comment:
I found the following logic slightly easier to understand (w/ other suggestions below):

```suggestion
                    delete_job = job.status.failed or job.status.succeeded
```